linker/shared/src/test/scala/org/scalajs/linker/testutils/TestIRRepo.scala

```diff
@@ -22,26 +23,39 @@ import org.scalajs.linker.irio._
 object TestIRRepo {
   val minilib = new TestIRRepo(StdlibHolder.minilib)
   val fulllib = new TestIRRepo(StdlibHolder.fulllib)
+
+  class InfoLoader(encodedNameToFile: Map[String, VirtualScalaJSIRFile]) {
+    private val infosCache = mutable.Map.empty[String, Future[ClassInfo]]
+
+    def loadInfo(encodedName: String)(
+        implicit ec: ExecutionContext): Option[Future[ClassInfo]] = {
+      infosCache.synchronized {
+        infosCache.get(encodedName).orElse {
```

Comment:
This never fills in the cache. Did you mean `getOrElseUpdate`?

Reply:
Hah, that seems to always have been the case :P Will fix.